File: Mahasiswa.java
/*Saya Muhamad Firdaus [2101995] mengerjakan
soal Latihan 2 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

// Deklarasi class human
class Human{
    // Atribut
    private String NIK;
    private String nama;
    private String jenis_kelamin;

    // Contructors tanpa parameter
    public Human(){
        this.NIK = "";
        this.nama = "";
        this.jenis_kelamin = "";
    }

    // Contructors dengan parameter
    public Human(String NIK, String nama, String jenis_kelamin){
        this.NIK = NIK;
        this.nama = nama;
        this.jenis_kelamin = jenis_kelamin;
    }

    // Get NIK
    public String getNIK(){
        return NIK;
    }

    // Set NIK
    public void setNIK(String NIK){
        this.NIK = NIK;
    }

    // Get Nama
    public String getNama(){
        return nama;
    }

    // Set Nama
    public void setNama(String nama){
        this.nama = nama;
    }

    // Get Jenis_kelamin
    public String getJenis_kelamin(){
        return jenis_kelamin;
    }

    // Set Jenis_kelamin
    public void setJenis_kelamin(String jenis_kelamin){
        this.jenis_kelamin = jenis_kelamin;
    }
}

// Deklarasi class SivitasAkademik extend class Human
class SivitasAkademik extends Human{
    // Atribut
    private String universitas;
    private String email_edu;

    // Constructors dengan parameter
    public SivitasAkademik(String NIK, String nama, String jenis_kelamin, String universitas, String email_edu){
        this.setNIK(NIK);
        this.setNama(nama);
        this.setJenis_kelamin(jenis_kelamin);
        this.universitas = universitas;
        this.email_edu = email_edu;
    }

    // Get Universitas
    public String getUniversitas(){
        return universitas;
    }

    // Set Universitas
    public void setUniversitas(String universitas){
        this.universitas = universitas;
    }

    // Get Email_edu
    public String getEmail_edu(){
        return email_edu;
    }

    // Set Email_edu
    public void setEmail_edu(String email_edu){
        this.email_edu = email_edu;
    }
}

// Deklarasi class SivitasAkademik extend class Mahasiswa
class Mahasiswa extends SivitasAkademik{
    // Atribut
    private String NIM;
    private String fakultas;
    private String prodi;

    // Constructors dengan parameter
    public Mahasiswa(String NIK, String nama, String jenis_kelamin, String universitas, String email_edu, String NIM, String fakultas, String prodi){
        // Menggunakan method super agar dapat mengakses Human
        super(NIK, nama, jenis_kelamin, universitas, email_edu);
        this.NIM = NIM;
        this.fakultas = fakultas;
        this.prodi = prodi;
    }

    // Get NIM
    public String getNIM(){
        return NIM;
    }

    // Set NIM
    public void setNIM(String NIM){
        this.NIM = NIM;
    }

    // Get Fakultas
    public String getFakultas(){
        return fakultas;
    }

    // Set Fakultas
    public void setFakultas(String fakultas){
        this.fakultas = fakultas;
    }

    // Get Prodi
    public String getProdi(){
        return prodi;
    }

    // Set Prodi
    public void setProdi(String prodi){
        this.prodi = prodi;
    }
}

File: python/program/Mahasiswa.py
# /*Saya Muhamad Firdaus [2101995] mengerjakan
# soal Latihan 2 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.*/

# Deklarasi class Human
class Human:
    # Atribut private
    __Nik = ""
    __nama = ""
    __jenis_kelamin= ""
    
    # Constructor
    def __init__(self, Nik="", nama="", jenis_kelamin=""):
        self.__Nik = Nik
        self.__nama = nama
        self.__jenis_kelamin = jenis_kelamin

    # Set NIK
    def setNik(self, Nik):
        self.__Nik = Nik

    # Get NIK
    def getNik(self):
        return self.__Nik

    # Set Nama
    def setNama(self, nama):
        self.__nama = nama

    # Get Nama
    def getNama(self):
        return self.__nama

    # Set Jenis Kelamin
    def setJenis_kelamin(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin

    # Get Jenis Kelamin
    def getJenis_kelamin(self):
        return self.__jenis_kelamin

# Deklarasi Sivitas Akademik
class SivitasAkademik(Human):
    # Atribut private
    __universitas = ""
    __email_edu = ""
    
    # Constructor
    def __init__(self, Nik="", nama="", jenis_kelamin="", universitas="", email_edu=""):
        super().__init__(Nik, nama, jenis_kelamin)
        self.__universitas = universitas
        self.__email_edu = email_edu

    # Set Universitas
    def setUniversitas(self, universitas):
        self.__universitas = universitas

    # Get Universitas
    def getUniversitas(self):
        return self.__universitas

    # Set email_edu
    def setEmail_edu(self, email_edu):
        self.__email_edu = email_edu

    # get email_edu
    def getEmail_edu(self):
        return self.__email_edu

# Deklarasi class Mahasiswa
class Mahasiswa(SivitasAkademik):
    # Atribut private
    __Nim = ""
    __fakultas = ""
    __prodi = ""
    
    # Constructor
    def __init__(self, Nik="", nama="", jenis_kelamin="", universitas="", email_edu="", Nim="", fakultas="", prodi=""):
        super().__init__(Nik, nama, jenis_kelamin, universitas, email_edu)
        self.__Nim = Nim
        self.__fakultas = fakultas
        self.__prodi = prodi

    # Set NIM
    def setNim(self, Nim):
        self.__Nim = Nim

    # Get NIM
    def getNim(self):
        return self.__Nim

    # Set Fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

    # Get Fakultas
    def getFakultas(self):
        return self.__fakultas

    # Set Prodi
    def setProdi(self, prodi):
        self.__prodi = prodi

    # Get Prodi
    def getProdi(self):
        return self.__prodi